internal/jobqueue/dbjobqueue/dbjobqueue.go

```diff
@@ -657,6 +657,46 @@ func (q *DBJobQueue) JobsUptoByType(jobTypes []string, upto time.Time) (result m
 	return
 }
 
+// Deletes single job and dependencies (recursively)
+// TODO remove transaction
+// TODO add jobqueue tests
+func (q *DBJobQueue) DeepDependencies(jobId uuid.UUID) ([]uuid.UUID, error) {
```

Comment:
So I'm unsure if using the recursive query is a good idea, might be more straightforward to just query the jobstatus while it has dependencies which errored out.

Reply:
Yeah it's a little messy. I'll get this split up in the meantime and get back to this